mscore/editstyle.cpp

```diff
@@ -676,6 +676,83 @@ EditStyle::EditStyle(Score* s, QWidget* parent)
       cs->startCmd();
       }
 
+//---------------------------------------------------------
+//   PAGES
+//---------------------------------------------------------
+
+const std::map<ElementType, int> EditStyle::PAGES = {
+      { ElementType::SCORE,               0  },
+      { ElementType::PAGE,                1  },
+      { ElementType::MEASURE_NUMBER,      4  },
+      { ElementType::BRACKET,             5  },
+      { ElementType::BRACKET_ITEM,        5  },
+      { ElementType::CLEF,                6  },
+      { ElementType::ACCIDENTAL,          7  },
+      { ElementType::MEASURE,             8  },
+      { ElementType::REST,                8  },
+      { ElementType::BAR_LINE,            9  },
+      { ElementType::NOTE,                10 },
+      { ElementType::CHORD,               10 },
+      { ElementType::STEM,                10 },
+      { ElementType::STEM_SLASH,          10 },
+      { ElementType::BEAM,                11 },
+      { ElementType::TUPLET,              12 },
+      { ElementType::ARPEGGIO,            13 },
+      { ElementType::SLUR,                14 },
+      { ElementType::SLUR_SEGMENT,        14 },
+      { ElementType::TIE,                 14 },
+      { ElementType::TIE_SEGMENT,         14 },
+      { ElementType::HAIRPIN,             15 },
+      { ElementType::HAIRPIN_SEGMENT,     15 },
+      { ElementType::VOLTA,               16 },
+      { ElementType::VOLTA_SEGMENT,       16 },
+      { ElementType::OTTAVA,              17 },
+      { ElementType::OTTAVA_SEGMENT,      17 },
+      { ElementType::PEDAL,               18 },
+      { ElementType::PEDAL_SEGMENT,       18 },
+      { ElementType::TRILL,               19 },
+      { ElementType::TRILL_SEGMENT,       19 },
+      { ElementType::VIBRATO,             20 },
+      { ElementType::VIBRATO_SEGMENT,     20 },
+      { ElementType::BEND,                21 },
+      { ElementType::TEXTLINE,            22 },
+      { ElementType::TEXTLINE_SEGMENT,    22 },
+      { ElementType::ARTICULATION,        23 },
+      { ElementType::FERMATA,             24 },
+      { ElementType::STAFF_TEXT,          25 },
+      { ElementType::TEMPO_TEXT,          26 },
+      { ElementType::LYRICS,              27 },
+      { ElementType::LYRICSLINE,          27 },
+      { ElementType::LYRICSLINE_SEGMENT,  27 },
+      { ElementType::DYNAMIC,             28 },
+      { ElementType::REHEARSAL_MARK,      29 },
+      { ElementType::FIGURED_BASS,        30 },
+      { ElementType::HARMONY,             31 },
```

Comment:
Could the elements perhaps be mapped by page names ([objectName](https://doc.qt.io/qt-5/qobject.html#objectName-prop) or something like that) rather than numbers? I'm afraid that in case of adding or reordering some pages in that dialog all this table will need manual editing. This doesn't look like a very interesting process to do :)

Reply:
yeah, that might be better. I'll do that when I have time.

Reply:
Done, I've moved all the page indices into an enum which should make things a lot easier to handle when reordering pages.

Reply:
Well, that's not quite what I meant but I have a (maybe) even better suggestion: it is actually possible not to search for the correct page by its `objectName` string but reference them by their object name directly. If we make this table an `EditStyle` member and initialize it in the constructor we can write something like this:
```
pages = {
      { ElementType::SCORE,             PageScore    },
      { ElementType::PAGE,              PagePage     },
// ...
      };
```

Those `PageScore` and other variables are just pointers to QWidgets representing those pages we want to display (these are just object names for those pages in the `.ui` file). However we will need to keep that page in sync with pages list, maybe the simplest solution for now will be just to keep using `setPage()` to switch to the desired page using [`pageStack->indexOf(page)`](https://doc.qt.io/qt-5/qstackedwidget.html#indexOf) to find the correct page index.

Of course, this solution pushes some (little) amount of calculations to be done at runtime but I see the following benefits from it:
1) No need to manually renumber/reorder anything in case of `editstyle.ui` changes.
2) Compile-time checking that all the necessary pages actually exist (if some page gets removed/renamed then the relevant variable pointing to that page's widget just will not exist).
3) No chance to make a mistake in page numbers (which is partially interconnected with item 1).

So what do you think on this option to implement linking the types to pages?

Reply:
That's a nice way of doing it. I'll look into implementing that.

Reply:
I think I've managed it. Could you review @dmitrio95 ?

Reply:
Looks great now! I'll try to test it in a few days, or perhaps @anatoly-os could do that if he manages to do it earlier.